CIPs/cip-137.md

```diff
@@ -18,187 +18,223 @@ An API for reading and writing events on Ceramic event streams.
+  - *eventId* <string> - the id of the event
+  - *id* <string> - CID of the InitEvent for this stream
+  - *prev* <array<string>> - CIDs of previous hash linked events
+  - *header* <object> - header for the event
+  - *data* <object> - the data of the event
+  - *timestamp* <integer> - the unixtime this event was timestamped (if it has been)
 
+**CAR file:**
 
+Content-Type:  `application/vnd.ipld.car`
 
-### `ceramic_reconPoll`
+This content types returns the underlaying IPLD data structure of the event streams encoded as a CAR file. This data can be used to trustlessly verify the integrity of the event stream. 
 
-Same as `ceramic_recon`, but waits for *duration* amount of time in case any new events arrives at the node during this time.
+The root object of the CAR file is *EventsResult* which is a convenience type containing an array of CIDs corresponding to the eventsids requested. The CAR file also includes all *Event* objects, including all IPLD blocks for these event, but no data from the previous events, e.g. DataEvents include signature envelope, event, and potentially detached payload, while TimeEvents include their entire merkle tree witness.
 
-**Params:**
+```verilog
+// https://developers.ceramic.network/protocol/streams/event-log/#data-event
+type Event InitEvent | DataEvent | TimeEvent
 
-- *reconRange* <Array<string>> - a Recon message, *`[eventid or ahash]`*
-- *duration* <integer> - the amount of time, in seconds, to wait for new data
+type Events [&Event]
+```
 
-**Returns:**
 
-- *reconRange* <Array<string>> - a Recon message, *`[eventid or ahash]`*
 
 ---
 
+### POST `/ceramic/events`
 
+Adds a new event to the node.
 
-### `ceramic_exportRawEvents`
+#### Request body:
 
-Get the raw IPLD data of a set of events given an array of eventids. This includes all IPLD blocks for this particular event, but no data from the previous events, e.g. DataEvents include signature envelope, event, and potentially detached payload, while TimeEvents include their entire merkle tree witness.
+The request body must contain the complete event data.
 
-**Params:**
+**CAR:**
 
-- *[eventid]* <array<string>> - the events to fetch
+Content-Type:  `application/vnd.ipld.car`
 
-**Returns:**
+A CAR file where the root is an *Event* as described below.
 
-- *events* <string> - a base64 encoded CAR file containing the event blocks
-
----
+```verilog
+// https://developers.ceramic.network/protocol/streams/event-log/#data-event
+type Event InitEvent | DataEvent | TimeEvent
+```
 
+**JSON:**
 
+Content-Type:  `application/json`
 
-### `ceramic_putRawEvent`
+A possible convenience method for adding an event by only submitting a JWT.
 
-Add an event to a stream.
+Only possible once we’ve migrated to use *Varsig* and can create events as plain signed JWTs.
 
-**Params:**
+```javascript
+{
+  event: "<jwt>"
+}
+```
 
-- *event* <string> - a base64 encoded CAR file containing the event
+#### Returns:
 
-**Returns:**
+**JSON:**
```

Comment:
Does this need a return? Should this just be a 200 if successful, error otherwise?

Reply:
Updated to return the eventId, as @PaulLeCam suggested.